org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigbeeIsAliveTracker.java

```diff
@@ -0,0 +1,83 @@
+package org.openhab.binding.zigbee.handler;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service which starts timeout tasks to set a Thing to OFFLINE if it doesn't reset the timer before it runs out
+ */
+@Component(immediate = true, service = ZigbeeIsAliveTracker.class)
+public class ZigbeeIsAliveTracker {
+
+    private final Logger logger = LoggerFactory.getLogger(ZigbeeIsAliveTracker.class);
+
+    private Map<ZigBeeThingHandler, Integer> handlerIntervalMapping = new ConcurrentHashMap<>();
+    private Map<ZigBeeThingHandler, ScheduledFuture<?>> scheduledTasks = new ConcurrentHashMap<>();
+
+    protected final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool("ZigbeeIsAliveTracker");
+
+    /**
+     * Adds a mapping from {@link ZigBeeThingHandler} to the interval in which it should have communicated
+     *
+     * @param zigBeeThingHandler      the {@link ZigBeeThingHandler}
+     * @param pollOrReportingInterval the interval in which the device should have communicated with us
+     */
+    public void addHandlerIntervalMapping(ZigBeeThingHandler zigBeeThingHandler, int pollOrReportingInterval) {
```

Comment:
Do these methods need to have the ```IntervalMapping``` on the end? Why not just ```addHandler``` and ```removeHandler```? The fact that it uses a map is implementation detail that doesn't really need to be reflected in the method name?

Reply:
I agree, will change it.